src/open_inwoner/js/components/FilterBar/multiselect_listbox_checkbox.js

```diff
@@ -2,134 +2,225 @@
 
-  // Display the sum
-  if (frequencySumElement) {
-    frequencySumElement.textContent = sum
-  }
+      if (listboxDropdown) {
+        listboxDropdown.addEventListener('keydown', (e) => {
+          const items = listboxDropdown.querySelectorAll(
+            '.filter-bar .checkbox__label'
+          )
+          if (e.key === 'ArrowDown') {
+            e.preventDefault()
+            currentIndex = (currentIndex + 1) % items.length
+            items[currentIndex].focus()
+          } else if (e.key === 'ArrowUp') {
+            e.preventDefault()
+            currentIndex = (currentIndex - 1 + items.length) % items.length
+            items[currentIndex].focus()
+          } else if (e.key === 'Escape') {
+            listboxDropdown.classList.remove('show')
+            selectButton.setAttribute('aria-expanded', 'false')
+            selectButton.focus()
+          }
+        })
+      }
 
-  // Ensure the filterCasesButton and filterFormActions exist before modifying them
-  if (filterCasesButton && filterFormActions) {
-    if (sum > 0) {
-      filterCasesButton.classList.remove('hide')
-      filterFormActions.classList.remove('hide')
-    } else {
-      filterCasesButton.classList.add('hide')
-      filterFormActions.classList.add('hide')
+      initCheckboxStateFromURL()
     }
-  }
-}
 
-// Listen for checkbox change events to update the sum
-document.addEventListener('change', function (e) {
-  if (e.target && e.target.classList.contains('checkbox__input')) {
-    calculateAndDisplayCheckedSum() // Update sum on toggle
-  }
-})
+    document.addEventListener('change', function (e) {
+      if (e.target && e.target.classList.contains('checkbox__input')) {
+        if (e.target.closest('.filter-bar')) {
+          calculateAndDisplayCheckedSum() // Update the display when a checkbox is checked/unchecked
+        }
+      }
+    })
 
-// Listen for the htmx:afterSwap event to initialize the select behavior and recalculate sum
-document.body.addEventListener('htmx:afterSwap', function () {
-  const listboxDropdown = document.getElementById('listboxDropdown')
-  const dataMultiSelectLabel = document.querySelectorAll('.checkbox__label')
+    // Reset button functionality
+    const resetFilters = document.getElementById('resetFilters')
+    if (resetFilters) {
+      resetFilters.addEventListener('click', function (e) {
+        e.preventDefault()
+
+        // Uncheck all checkboxes
+        const checkboxes = document.querySelectorAll(
+          '.filter-bar .checkbox__input'
+        )
+        checkboxes.forEach((checkbox) => {
+          checkbox.checked = false
+        })
+
+        calculateAndDisplayCheckedSum() // Update the button and reset state after clearing
+
+        // Optionally, resubmit the form after resetting filters,
+        // may need to be added when submit button is removed in future
+        // const form = document.querySelector('#filterBar .form')
+        // if (form) {
+        //   form.submit()
```

Comment:
@pi-sigma  I kept this out for now, but see how you feel about this, once you start interacting with the component.

Reply:
Currently there is no way to re-select all aanvragen after filtering through the form (you have to deselect the filters and then refresh the page). So this should definitely be added.

Alternatively, always show the form button. In this case, the button also needs the correct number (currently it has "Toon 0 resultaten" if I manually change the relevant conditional in `multiselect_listbox_checkbox.js`), or the text could just be "Toon alle resultaten" if no filter is selected.

The `console.log` from below though can be deleted of course